### PR TITLE
Fix deprecated oneMKL call

### LINKED ISSUE #1717: Update for MKL 2025
MKL single argument release_matrix_handle was deprecated in 2023 and removed in 2025
https://github.com/ginkgo-project/ginkgo/blob/568a75946d09218e57492da18da552267ab8b3da/dpcpp/matrix/csr_kernels.dp.cpp#L1420

https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-dpcpp/2025-0/oneapi-mkl-sparse-release-matrix-handle.html

We are trying to release oneapi 2025 on Spack, but e4s build is failing: 
https://github.com/spack/spack/pull/47264
with this error
```
/tmp/root/spack-stage/spack-stage-ginkgo-1.8.0-bqbygy3vpm73xu5bswwd4eo7ekitjsrm/spack-src/dpcpp/matrix/csr_kernels.dp.cpp:1424:63: error: too few arguments to function call, expected at least 2, have 1
 1424 |         oneapi::mkl::sparse::release_matrix_handle(&mat_handle);
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~            ^
/home/software/spack/[padded-to-256-chars]/linux-ubuntu22.04-x86_64_v3/oneapi-2024.2.1/intel-oneapi-mkl-2025.0.0-trzhgb6lg67dxvpj47qoycibohcjnzus/mkl/latest/include/oneapi/mkl/spblas/sparse_structures.hpp:41:1: note: 'release_matrix_handle' declared here
   41 | release_matrix_handle(sycl::queue &queue,
      | ^                     ~~~~~~~~~~~~~~~~~~~
   42 |                       matrix_handle_t *p_spMat,
      |                       ~~~~~~~~~~~~~~~~~~~~~~~~~
   43 |                       const std::vector<sycl::event> &dependencies = {});
```

There might be more deprecation/removal issues. Would it be possible to update Ginkgo?

### PR DESCRIPTION
**Pull request by MarcelKoch**
This PR uses the new overload of `release_matrix_handle` and `set_csr_data` instead of the deprecated one, see https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-dpcpp/2025-0/oneapi-mkl-sparse-release-matrix-handle.html and https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-dpcpp/2025-0/oneapi-mkl-sparse-set-csr-data.html
Also, the member functions for shuffle in subgroup and `is_host()` are removed in the oneapi 2025.0.0
Use the group algorithm and remove `is_host()` in this pr.
The group algorithm should still be supported from 2023.1

Fixes #1717 

Todo:
- [x] wait until intel CI is running again.
- [x] check for other removed oneAPI calls.

### CODE REVIEW

**Comment by yhmtsai** [comment]
ah, you are also working on that.
you need to change the set_csr_data, too.

**Comment by yhmtsai** [comment]
I am preparing the docker image such that we can check the compilation at least

**Comment by MarcelKoch** [comment]
@yhmtsai feel free to push if you already have a fix for set_csr_data.

**Comment by MarcelKoch** [comment]
format!

**Comment by MarcelKoch** [comment]
@pratikvn We currently test 2023.1 and 2024.1 (this is the `oneapi:latest` image). For me that is good enough, but we can also try to add 2025 if there is enough time before the release.

**Review by upsj — APPROVED** [review]
LGTM!

File: benchmark/utils/dpcpp_linops.dp.cpp [hunk 1]
```diff
@@ -47,12 +47,16 @@ class OnemklBase : public gko::LinOp {
...
     {
         mat_handle_ = handle_manager<oneapi::mkl::sparse::matrix_handle>(
             create_mat_handle(),
-            [](oneapi::mkl::sparse::matrix_handle_t mat_handle) {
-                oneapi::mkl::sparse::release_matrix_handle(&mat_handle);
+            [exec](oneapi::mkl::sparse::matrix_handle_t mat_handle) {
+                oneapi::mkl::sparse::release_matrix_handle(
+#if INTEL_MKL_VERSION >= 20240000
+                    *exec->get_queue(),
+#endif
```
[the remark this answers is not in the document]
Comment (MarcelKoch) [reply benchmark/utils/dpcpp_linops.dp.cpp:58]: Yes, but I think the new one doesn't work in 2023.1. At least running on the PVC@FTP it always segfaulted.

[the remark this answers is not in the document]
Comment (yhmtsai) [reply benchmark/utils/dpcpp_linops.dp.cpp:58]: I see. Thanks for clarifying it

**Pull request merged**